packages/aws-cdk/test/api/exec.test.ts

```diff
@@ -0,0 +1,126 @@
+import * as cxschema from '@aws-cdk/cloud-assembly-schema';
+import * as cdk from '@aws-cdk/core';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as semver from 'semver';
+import { ImportMock } from 'ts-mock-imports';
+import { SdkProvider } from '../../lib';
+import { execProgram } from '../../lib/api/cxapp/exec';
+import { Configuration, Settings } from '../../lib/settings';
+
+// We need to increase the default 5s jest
+// timeout for async tests because the 'execProgram' invocation
+// might take a while :\
+const TEN_SECOND_TIMEOUT = 10000;
+
+function createApp(outdir: string): cdk.App {
+
+    const app = new cdk.App({outdir});
+    const stack = new cdk.Stack(app, 'Stack');
+
+    new cdk.CfnResource(stack, "Role", {
+        type: "AWS::IAM::Role",
+        properties: {
+            RoleName: "Role"
+        }
+    });
+
+    return app;
+}
+
+async function createSdkProvider() {
+    return await SdkProvider.withAwsCliCompatibleDefaults();
+}
+
+test('execProgram throws when manifest version > schema version', async () => {
+
+    const outdir = fs.mkdtempSync(path.join(os.tmpdir(), 'exec-tests'));
+
+    const app = createApp(outdir);
+
+    const currentSchemaVersion = cxschema.Manifest.version();
+
+    const mockManifestVersion = semver.inc(currentSchemaVersion, 'major');
+
+    // this mock will cause the framework to use a greater schema version than the real one,
+    // causing the cli to fail (hopefully).
+    const mockVersionNumber = ImportMock.mockFunction(cxschema.Manifest, 'version', mockManifestVersion);
+    try {
+        app.synth();
+    } catch (err) {
+        fail(err.message);
+    } finally {
+        mockVersionNumber.restore();
+    }
```

Comment:
Use jest

```suggestion
    expect(() => app.synth()).not.toThrow();
    mockVersionNumber.restore();
```

Reply:
But this won't restore the mock if `synth` actually does throw. Also im not really attempting to do any assertions here, i just wanted to validate the mock is restored, i think just:

```typescript
try {
  app.synth()
} finally {
  mockVersionNumber.restore()
}
```

Not sure why I explicitly failed the test in the `catch` block.